### PR TITLE
Support custom PK algo types

### PR DESCRIPTION
**Pull request by jack-fortanix**
Same as internal version

### CODE REVIEW

File: mbedtls/src/ssl/ciphersuites.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@
  * according to those terms. */
 
 use mbedtls_sys::types::raw_types::c_int;
+use mbedtls_sys::*;
```
Comment (Goirad) [inline mbedtls/src/ssl/ciphersuites.rs]: Why is this import here if nothing is being added to the file?

**Comment by Goirad** [comment]
Also it would be nice to have one more PR before doing a release that just updates everything for 2018 edition and resolves deprecation warnings

File: mbedtls/src/ssl/ciphersuites.rs (hunk 1 above)
Comment (jack-fortanix) [reply mbedtls/src/ssl/ciphersuites.rs]: I had to change `define_enum` macro to not assume the values are in `mbedtls_sys::` since we need to reference a value that is not defined in that crate (`CUSTOM_PK_TYPE`)

Comment (Goirad) [reply mbedtls/src/ssl/ciphersuites.rs]: I see

**Comment by Goirad** [comment]
Oh it looks like there are some conflicts

**Comment by jack-fortanix** [comment]
@Goirad I will create a PR updating edition
